src/types/box.js

```diff
@@ -109,31 +97,65 @@ BoxAnnotation.descriptors = {
   }
 };
 
-function calculateX(box, labelSize, position, padding) {
-  const {x: start, x2: end, width: size, options} = box;
-  const {xAdjust: adjust, borderWidth} = options.label;
-  return calculatePosition({start, end, size}, {
+function calculateX({properties, options}, labelSize, position, padding) {
+  const {x: start, x2: end, width: size} = properties;
+  return calculatePosition({start, end, size, borderWidth: options.borderWidth}, {
     position: position.x,
     padding: {start: padding.left, end: padding.right},
-    adjust, borderWidth,
+    adjust: options.label.xAdjust,
     size: labelSize.width
   });
 }
 
-function calculateY(box, labelSize, position, padding) {
-  const {y: start, y2: end, height: size, options} = box;
-  const {yAdjust: adjust, borderWidth} = options.label;
-  return calculatePosition({start, end, size}, {
+function calculateY({properties, options}, labelSize, position, padding) {
+  const {y: start, y2: end, height: size} = properties;
+  return calculatePosition({start, end, size, borderWidth: options.borderWidth}, {
     position: position.y,
     padding: {start: padding.top, end: padding.bottom},
-    adjust, borderWidth,
+    adjust: options.label.yAdjust,
     size: labelSize.height
   });
 }
 
 function calculatePosition(boxOpts, labelOpts) {
-  const {start, end} = boxOpts;
-  const {position, padding: {start: padStart, end: padEnd}, adjust, borderWidth} = labelOpts;
+  const {start, end, borderWidth} = boxOpts;
+  const {position, padding: {start: padStart, end: padEnd}, adjust} = labelOpts;
   const availableSize = end - borderWidth - start - padStart - padEnd - labelOpts.size;
-  return start + borderWidth / 2 + adjust + padStart + getRelativePosition(availableSize, position);
+  return start + borderWidth / 2 + adjust + getRelativePosition(availableSize, position);
+}
+
+function getBox(properties, padding, borderWidth) {
+  const {x, y, width, height} = properties;
+  const halfBorder = borderWidth / 2;
+  return {
+    x: x + halfBorder + padding.left,
+    y: y + halfBorder + padding.top,
+    width: width - borderWidth - padding.width,
+    height: height - borderWidth - padding.height
+  };
+}
+
+function resolveLabelElementProperties(chart, properties, options) {
+  const label = options.label;
+  label.backgroundColor = 'transparent';
+  label.callout.display = false;
```

Comment:
added these properties settings to override whatever user configuration (box label should work without background and callout)

Reply:
Ref: https://github.com/chartjs/chartjs-plugin-annotation/pull/698#discussion_r815340691

The question is, do we want to (force) the box label to not have background or callout? there could be a use case for those too. I think setting the defaults would be enough, if user overrides, then fine!

Reply:
Well, the box label wasn't thought to use callout and background color.
Let me think about... for callout I need to think about a point to link to the callout (could be the center of the label or top/left of the label). The box annotation (but the same for line) we don't have any option to locate a point (correctly).